zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterContext.java

```diff
@@ -27,6 +27,21 @@
  * Interpreter context
  */
 public class InterpreterContext {
+  private static final ThreadLocal<InterpreterContext> threadIC =
```

Comment:
I'm not sure if i understand your suggestion correctly, but InterpreterContext is created every time. For example, new instance of InterpreterContext is created passed into Interpreter.interpret() method. So i think it's hard to make InterpreterContext a singleton.